docs/src/extend/config-management.md

```diff
@@ -0,0 +1,255 @@
+---
+title: Config Management
+tags: [Extend, Configuration Management]
+sidebar_position: 3
+description: Extend configuration management capabilities with plugins
+---
+
+This document describes how to extend the configuration management capabilities of %%te%% using plugins.
+
+By default, %%te%% supports managing file-based configurations.
+But, this support is limited to reading or updating a file on the file system.
+
+It has the following limitations:
+1. Non-file based configuration sources are not supported
+2. While updating a configuration, no pre-processing or post-processing of the config can be performed.
+
+Custom configuration plugins address these limitations and enable more flexible configuration workflows.
+
+## Overview
+
+The %%te%% agent supports extensible configuration management through a plugin system:
+
+* Plugins can be installed to support any kind of configuration sources, including non-file formats
+* Each plugin can handle multiple configuration types
+* Plugins are discovered and executed automatically by `tedge-agent`
+
+## How Config Plugins Work
+
+A config plugin is an executable that implements a simple command-line interface with the following requirements:
+
+* Implement three sub-commands:
+  1. **`list`** - Returns all config types supported by the plugin (one per line)
+  2. **`get <type>`** - Retrieves the current configuration for the specified type and outputs it to stdout
+  3. **`set <type> <new-config-path>`** - Updates the configuration for the specified type from the provided config file path
+* Must exit with code 0 for successful command (used to validate the plugin using the `list` command).
+* Should output the configuration content to stdout for the `get` command.
+* Should handle validation, backup, and restoration in the `set` command.
+* Should handle errors gracefully and exit with non-zero codes on failure.
+
+The agent automatically:
+* Discovers plugins at startup by running their `list` command to gather their supported config types.
```

Comment:
I would move here the note about the list command exit code.

```suggestion
* Discovers plugins at startup by running their `list` command to gather their supported config types (skipping plugins returning a non zero exit code on `list`).
```

Reply:
Resolved by 7e4c33a